src/main/scala/viper/silver/plugin/standard/termination/TerminationPlugin.scala

```diff
@@ -106,6 +115,75 @@ class TerminationPlugin(@unused reporter: viper.silver.reporter.Reporter,
     newProgram
   }
 
+  override def beforeTranslate(input: PProgram): PProgram = {
+    val allClauseTypes: Set[Any] = decreasesClauses.flatMap{
+      case PDecreasesTuple(Seq(), _) => Seq(())
+      case PDecreasesTuple(exps, _) => exps.map(e => e.typ match {
+        case PUnknown() if e.isInstanceOf[PCall] => e.asInstanceOf[PCall].idnuse.typ
+        case _ => e.typ
+      })
+      case _ => Seq()
+    }.toSet
+    val presentDomains = input.domains.map(_.idndef.name).toSet
+
+    val importStmts = allClauseTypes flatMap {
+      case TypeHelper.Int if !presentDomains.contains("IntWellFoundedOrder") => Some("import <decreases/int.vpr>")
+      case TypeHelper.Ref if !presentDomains.contains("RefWellFoundedOrder") => Some("import <decreases/ref.vpr>")
+      case TypeHelper.Bool if !presentDomains.contains("BoolWellFoundedOrder") => Some("import <decreases/bool.vpr>")
+      case TypeHelper.Perm if !presentDomains.contains("RationalWellFoundedOrder") => Some("import <decreases/rational.vpr>")
+      case PMultisetType(_) if !presentDomains.contains("MultiSetWellFoundedOrder") => Some("import <decreases/multiset.vpr>")
+      case PSeqType(_) if !presentDomains.contains("SeqWellFoundedOrder") => Some("import <decreases/seq.vpr>")
+      case PSetType(_) if !presentDomains.contains("SetWellFoundedOrder") => Some("import <decreases/set.vpr>")
+      case PPredicateType() if !presentDomains.contains("PredicateInstancesWellFoundedOrder") =>
+        Some("import <decreases/predicate_instance.vpr>")
+      case _ if !presentDomains.contains("WellFoundedOrder") => Some("import <decreases/declaration.vpr>")
```

Comment:
It seems your code assumes that e.g. in the case of `Int`, user provided custom well founded orders will be provided in a domain called "IntWellFoundedOrder", right?
However, the tutorial imho does not mandate a particular domain name for providing a custom well founded order. Thus, I believe your code would end up importing the default well-founded order axioms, which would easily lead to an unsoundness (probably even in most cases as there was a reason to provide a custom well-founded order)

Reply:
Makes sense. I prefer this solution over an IDE integration but still, I'm a bit concerned that Viper silently includes two additional axioms that result in contradictions with the axioms a user has defined for the custom well-founded order.
Could we check whether axioms refer to `decreases` and `bounded` and maybe emit a warning that e.g. they are not within a domain of expected name?

Reply:
As discussed, we now check all domains that are already in the programs for axioms that constrain ``bounded`` and ``decreases`` for any type ``T``. If such a constraint is found in a domain that's not called ``TWellFoundedOrder``, we emit a warning.